src/Autocomplete.js

```diff
@@ -25,6 +25,16 @@ class Autocomplete extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.instance) {
+      const { data } = this.props.options;
+
+      if (prevProps.options.data !== data) {
```

Comment:
```js
var a = { foo: 2, baz: 3 }
var b = { baz: 3, foo: 2 }
a === b // false
```

I think it's **ok** to `updateData` every time here

Reply:
What if we use this method?

```
var a = { foo: 2, baz: 3 }
var b = { baz: 3, foo: 2 }

JSON.stringify(a) === JSON.stringify(b) //false (the order matters)
JSON.stringify(a) === JSON.stringify({ foo:2, baz: 3}) //true
```

Concerning performance it's OK if the object is relatively small.
Moreover this also works  for deep nested objects.

Considering all of this and that `data` object is at most of the form 
```
{
     "Apple": null,
     "Microsoft": null,
     "Google": 'https://placehold.it/250x250'
}
```
 it should be OK.

Let me know your thoughts :)

Reply:
I've also tested this approach with a `spec`. It seems to work just fine.
For now I'm going to keep it on my machine until you provide me with some feedback.
Or would you like me to push this modification? :relaxed:

Reply:
I'm afraid we can't assume the order of an object, I would rather remove the checking

Reply:
We can't just always call `updateData`, otherwise the component is not going to properly work.
It would call `updateData` even when the user is typing.
I'd really be grateful if you could have a look at the code I've now

Reply:
ping @alextrastero

Reply:
Sorry for the delay. We need to solve this in a way that if we supply the same object but with different order we don't want to call `updateData` correct?